### PR TITLE
Add page *Release Policy*

### PR DESCRIPTION
**Pull request by straight-shoota**
Resolves https://github.com/crystal-lang/crystal/issues/14481

The content is heavily based on [hexdocs.pm/elixir/compatibility-and-deprecations.html](https://hexdocs.pm/elixir/compatibility-and-deprecations.html), which is a great summary especially concerning backwards compatibility and matches the intentions of the Crystal project (as far as I'm concerned).

### CODE REVIEW

File: docs/project/release-policy.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+Although we expect the vast majority of programs to remain compatible over time,
+it is impossible to guarantee that no future change will break any program.
+Under some unlikely circumstances, we may introduce changes that break existing code.
+Rest assured we are commited to keep the impact as minimal as possible.
+
+* Security: a security issue in the implementation may arise whose resolution requires backwards incompatible changes. We reserve the right to address such security issues.
+
+* Bugs: if an API has undesired behaviour, a program that depends on the buggy behaviour may break if the bug is fixed. We reserve the right to fix such bugs.
+
+* Compiler front-end: improvements may be done to the compiler, introducing new warnings for ambiguous modes and providing more detailed error messages. Those can lead to compilation errors (when building with `--error-on-warnings`) or tooling failures when asserting on specific error messages (although one should avoid such). We reserve the right to do such improvements.
```
[the remark this answers is not in the document]
Comment (straight-shoota) [reply docs/project/release-policy.md:34]: Not sure what you mean. Could you give an example?
In general, I'd assume that refactorings in the compiler should uphold compatibility.

[the remark this answers is not in the document]
Comment (Blacksmoke16) [reply docs/project/release-policy.md:34]: Sure, for example https://github.com/crystal-lang/crystal/pull/12903. It did some internal breaking changes which ended up being the cause of https://github.com/olbat/libgen/issues/70. 

I was proposing making it explicit that the backwards compatibility promises do not cover `src/compiler`, even if _most_ changes are compatible.

[the remark this answers is not in the document]
Comment (straight-shoota) [reply docs/project/release-policy.md:34]: Ah, you mean using the compiler API directly. That's enirely on the consumer's risk. These are undocumented internal APIs which are not part of the compatibility guarantee.

File: docs/README.md [hunk 2]
```diff
@@ -114,11 +114,12 @@ Instructions on how to use the compiler and tools.
   </div>
   <div class="card" markdown="1">
 
-### Developer News
+### Project News
 
 Announcements about the language development.
 
 * [Release Notes](https://crystal-lang.org/blog/#release_notes)
```
Comment (beta-ziliani) [inline docs/README.md:121]: Fixup

```suggestion
* [Release Notes](https://crystal-lang.org/releases)
```

File: docs/project/release-policy.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (straight-shoota) [reply docs/project/release-policy.md:34]: > Well-defined behaviours and documented APIs in a given version
will continue working on future versions within the same major branch.

IMO this clearly exculdes undocumented and internal APIs 🤷

[the remark this answers is not in the document]
Comment (beta-ziliani) [reply docs/project/release-policy.md:34]: What constitutes "Well-defined behaviours"?

[the remark this answers is not in the document]
Comment (straight-shoota) [reply docs/project/release-policy.md:34]: This seems like a different topic, isn't it? I'm happy to discuss the meaning of "well-defined" but does it relate to this? I was more focusing on the "documented APIs" part.

[the remark this answers is not in the document]
Comment (beta-ziliani) [reply docs/project/release-policy.md:34]: Perhaps this will suffice as a proof that we need to expand this? :-D

File: docs/project/release-policy.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+Patch releases contain only important bug fixes and are released when necessary.
+They usually only appear for the latest minor branch.
+
+New releases are announced at [crystal-lang.org/releases](https://crystal-lang.org/releases) ([RSS feed](https://crystal-lang.org/releases)).
+
+There are currently no plans for a new major release.
+
+## Backwards compatibility
+
+Minor and patch releases are backwards compatible: Well-defined behaviours and documented APIs in a given version
```
Comment (beta-ziliani) [inline docs/project/release-policy.md:18]: How about:

```suggestion
Minor and patch releases are backwards compatible: any use of documented API from the standard library is expected to work in future versions within the same major branch. The same is expected for undocumented but not hidden API (that is, public and without `:nodoc:`) and any language construct. There are no guarantees over any other aspect, like compiler internals or `:nodoc:` API.
```

Comment (straight-shoota) [reply docs/project/release-policy.md:18]: This feels a bit too wordy and thus hard to grasp. I'll try to rephrase this with fewer words.

File: docs/project/release-policy.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+will continue working on future versions within the same major branch.
```
Comment (straight-shoota) [inline docs/project/release-policy.md:19]:
```suggestion
Minor and patch releases are backwards compatible: Well-defined behaviours and documented APIs in a given version
will continue working on future versions within the same major branch.
This applies to all features listed in the [API docs](https://crystal-lang.org/api) and [language reference](https://crystal-lang.org/reference/syntax_and_semantics/). It covers any observable behaviour (unless its obviously invalid, i.e. a bug), even if the documentation does not explicitly mention it (i.e. the documentation is incomplete).
```

Comment (beta-ziliani) [reply docs/project/release-policy.md:19]: I prefer mine, but ok :-D (behavior if you prefer US english)

**Review by beta-ziliani — APPROVED** [review]
